nntrainer/tensor/tensor_pool.h

```diff
@@ -340,18 +299,35 @@ class TensorPool {
                         unsigned int offset);
 
 private:
+  /**
+   * @brief Source tensor detailed specification
+   *
+   */
+  struct SourceDetails {
+    unsigned int token;                   /**< memory token */
+    TensorLifespan lifespan;              /**< life span of the tensor */
+    std::vector<unsigned int> exec_order; /**< exec order */
+    std::vector<unsigned int>
+      dependents; /**< list of dependents to the source */
+  };
+
+  /**
+   * @brief Dependent tensor detaild specification
+   *
+   */
+  struct DependentDetails {
+    unsigned int parent_idx; /**< index to the parent */
+    unsigned int offset;     /**< elementwise offset */
```

Comment:
I highly recommend keeping all the details of the tensor request even if the tensor is dependent.
This is crucial to be able to modify an already sent request which is needed for graph optimization. In this design, it is impossible.

Reply:
Could you elaborate on why and what will need to be shared from source details, if there is any, I think it'd better be a  member of `RequestSpec` than removing `std::variant<DependentDetails, SourceDetails>`.

If usecase is limited to the modifying an already sent request, it must be done to the source tensor not to the requested dependent tensor to prevent invariant breaks. 

fyi) there is a proposed method for the suggested usecase `reidentifySourceTensor`.

Reply:
As per yesterday's offline discussion, reidentifing only happens to the source. So we clarified this can be added once we need it. Please correct me if this is wrong :)

Reply:
Yes, as per the offline discussion, the current use case is reidentifying for the source tensor only, so this works for now, and details for dependent tensors can be kept if use case arises.